File: Stratego Game/Week 1 iLab/Player.cpp
#include "Player.h"
#include "GamePiece.h"

Player::Player()
{
	for (int i = 0; i < 40; i++)
	{
		piece[i].init(0, -1);
		displayColor = 0;
	}
}

Player::Player(const Player & p)
{
	for (int i = 0; i < 40; i++)
	{
		piece[i] = p.piece[i];
	}
	displayColor = p.displayColor;
}

Player::~Player()
{
}

void Player::init(int c)
{
	if (c == 1)
	{
		displayColor = 0;
	}
	else if (c == 2)
	{
		displayColor = 1;
	}
}

Player & Player::operator = (const Player & p)
{
	for (int i = 0; i < 40; i++)
	{
		piece[i] = p.piece[i];
	}
	displayColor = p.displayColor;
	return *this;
}

GamePiece * Player::getGamePiece(int i)
{
	return &piece[i];
}

void Player::setGamePiece(int i, GamePiece * g)
{
	piece[i] = *g;
}

int Player::getColor()
{
	return displayColor;
}

void Player::setColor(int c)
{
	if (c == 1)
	{
		displayColor = 0;
	}
	else
	{
		displayColor = 1;
	}
}

File: Java Version/Player.java
package StrategoJava;

import java.awt.Graphics;

public class Player {
	private GamePiece pieces[];
	private int displayColor;
	
	public Player()
	{
		pieces = new GamePiece[40];
		for (int i = 0; i < 40; ++i)
		{
			pieces[i] = new GamePiece();
		}
		displayColor = 0;
	}
	
	void init (int c)
	{
		if (c == 1)
		{
			displayColor = 0;
		}
		else
		{
			displayColor = 1;
		}
	}
	
	public void setGamePiece (int i, GamePiece g)
	{
		pieces[i] = g;
	}
	
	public GamePiece getGamePiece(int i)
	{
		return pieces[i];
	}
	
	public void setColor (int c)
	{
		displayColor = c;
	}
	
	public int getColor ()
	{
		return displayColor;
	}
	
	public void draw(Graphics g)
	{
		for (int i = 0; i < 40; ++i)
		{
			pieces[i].draw(g);
		}
	}
	
	public boolean checkForCollision(Vector2D p)
	{
		for (int i = 0; i < 40; ++i)
		{
			if ((pieces[i].getPosition().getX() == p.getX()) && (pieces[i].getPosition().getY() == p.getY()))
			{
				return true;
			}
		}
		return false;
	}
}
